.github/workflows/tests.yml

```diff
@@ -39,61 +44,105 @@ jobs:
     outputs:
       full: ${{ steps.filter.outputs.full }}
 
-  pre-commit:
-    needs: changes
+  # pre-commit:
+  #   needs: changes
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: actions/setup-python@v5
+  #       with:
+  #         python-version: "3.x"
+  #     - name: Install tofu
+  #       run: |
+  #         curl --proto '=https' --tlsv1.2 -fsSL https://get.opentofu.org/install-opentofu.sh -o install-opentofu.sh
+  #         chmod +x install-opentofu.sh
+  #         ./install-opentofu.sh --install-method deb
+  #         rm -f install-opentofu.sh
+  #     - shell: bash
+  #       run: |
+  #         curl -fsSL https://deno.land/install.sh | sh -s v2.1.4
+  #         export DENO_INSTALL="$HOME/.deno"
+  #         export PATH="$DENO_INSTALL/bin:$PATH"
+  #         # ghjk p resolve
+  #         python -m pip install --upgrade pip
+  #         pip install pre-commit
+  #         pre-commit install
+  #         deno --version
+  #         pre-commit run --all-files
+  #         # SKIP=ghjk-resolve pre-commit run --show-diff-on-failure --color=always --all-files
+
+  # build-docker:
+  #   needs: changes
+  #   if: ${{ needs.changes.outputs.full == 'true' && !(github.event_name == 'workflow_dispatch' && inputs.tmate_enabled) && github.event.pull_request.draft == false  }}
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: docker/setup-buildx-action@v3
+  #     - uses: docker/login-action@v3
+  #       with:
+  #         registry: ghcr.io
+  #         username: ${{ github.actor }}
+  #         password: ${{ secrets.GITHUB_TOKEN }}
+  #     - name: Sanitize Docker tag
+  #       id: vars
+  #       run: |
+  #         if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+  #           REF="${{ github.head_ref }}"
+  #         else
+  #           REF="latest"
+  #         fi
+  #         echo "TAG=$(echo "$REF" | tr '/' '-')" >> $GITHUB_ENV
+  #     - name: Build and push herald
+  #       uses: docker/build-push-action@v6
+  #       with:
+  #         context: .
+  #         file: ./Dockerfile
+  #         platforms: linux/amd64,linux/arm64
+  #         push: true
+  #         tags: ghcr.io/${{ github.repository_owner }}/herald:${{ env.TAG }}
+
+  test-full:
+    needs: [changes]
+    if: ${{ needs.changes.outputs.full == 'true' && !(github.event_name == 'workflow_dispatch' && inputs.tmate_enabled) && github.event.pull_request.draft == false }}
```

Comment:
tmate_enabled? let's simplify from where you took that so it becomes as simple as possible